tensorflow/contrib/android/java/org/tensorflow/contrib/android/TensorFlowInferenceInterface.java

```diff
@@ -50,6 +56,117 @@
   private static final String TAG = "TensorFlowInferenceInterface";
   private static final String ASSET_FILE_PREFIX = "file:///android_asset/";
 
+  /**
+   * Copies resource to a cache file.
+   * @param  context context The context that contains the files
+   * @param  String filename  Filename--take note that it checks for .pb in asset case
+   * @param  boolean res Is it from res/raw
+   *
+   * @return File reference to cache file.
+   */
+  private File createCacheFile(Context context, String filename, boolean res)
+      throws IOException {
+    InputStream inputStream = null;
+    if (res) {
+      int resourceId = context.getResources().getIdentifier(filename, "raw", context.getPackageName());
+      if (resourceId == 0) {
+        return null;
+      }
+      inputStream = context.getResources().openRawResource(resourceId);
+    } else {
+      if (!filename.endsWith(".pb")) {
+        filename += ".pb";
+      }
+      inputStream = context.getAssets().open(filename);
+    }
+    File cacheFile = new File(context.getCacheDir(), filename);
+    if (cacheFile.createNewFile() == false) {
+      cacheFile.delete();
+      cacheFile.createNewFile();
+    }
+    FileOutputStream fileOutputStream = new FileOutputStream(cacheFile);
+    byte[] buffer = new byte[1024 * 512];
```

Comment:
Any particular reason for this number? If so, please comment.

Reply:
@andrewharp For Android, it might be a bit large.  I see that the cache is recommended to max out at 1MB.  I guess I should change it to use internal storage.